ods_ci/tasks/Resources/RHODS_OLM/pre-tasks/oc_is_operator_installed.robot

```diff
@@ -1,38 +1,86 @@
+              ...  Run Keywords
+              ...  Is RHODS Family Operator Installed  namespace=redhat-ods-operator  subscription=rhods-operator
+              ...          pod_selector=name=rhods-operator    AND
+              ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-monitoring  AND
+              ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-applications  AND
+              ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-operator  AND
+              ...  Oc Get  kind=CatalogSource  namespace=openshift-marketplace
+              ...          field_selector=metadata.name=redhat-operators
+          ELSE
+              FAIL    Provided test environment and install type combination is not supported
+          END
       END
   ELSE IF  "${cluster_type}" == "managed"
       IF  "${UPDATE_CHANNEL}" == "odh-nightlies"
           ${result}=  Run Keyword And Return Status
           ...  Run Keywords
-          ...  Oc Get  kind=OdhDashboardConfig  namespace=opendatahub  AND
+          ...  Is RHODS Family Operator Installed  namespace=opendatahub  subscription=addon-managed-odh
+          ...          pod_selector=name=rhods-operator    AND
           ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-operator
       ELSE
           ${result}=  Run Keyword And Return Status
           ...  Run Keywords
-          ...  Oc Get  kind=OdhDashboardConfig  namespace=redhat-ods-applications  AND
+          ...  Is RHODS Family Operator Installed  namespace=redhat-ods-operator  subscription=addon-managed-odh
+          ...          pod_selector=name=rhods-operator    AND
           ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-monitoring  AND
           ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-applications  AND
           ...  Oc Get  kind=Namespace  field_selector=metadata.name=redhat-ods-operator  AND
           ...  Oc Get  kind=CatalogSource  namespace=redhat-ods-operator
           ...          field_selector=metadata.name=addon-managed-odh-catalog
       END
+  ELSE
+      FAIL    Provided test environment and install type ${INSTALL_TYPE} ${UPDATE_CHANNEL} ${cluster_type} combination is not supported
   END
+  Log   RHODS is installed result: ${result}      console=yes
+  RETURN  ${result}
+
+Is RHODS Family Operator Installed
+  [Documentation]   Returns if an operator with given pod label selector is currently installed in given namespace
+  [Arguments]    ${namespace}  ${subscription}  ${pod_selector}
+  Log   Getting CSV from subscription ${subscription} namespace ${namespace}      console=yes
+  ${rc}    ${current_csv_name} =    Run And Return Rc And Output
+  ...    oc get subscription ${subscription} -n ${namespace} -ojson | jq '.status.currentCSV' | tr -d '"'
+  Log   Got CSV ${current_csv_name} from subscription ${subscription}, result: ${rc}      console=yes
```

Comment:
you could reuse this kw: `Wait Until Operator Subscription Last Condition Is`

Reply:
Good point! Just one doubt, is it ok to wait in this case? We are checking that the operator is installed or not (typically from a previous job run on the cluster), so waiting might not make sense in this case and just prolong the job duration?

On the other hand, it might help in some corner case, when the operator would have been installed/installing at the time this RHODS operator installation check was being made.. not sure if that is expected, though.

If you feel like reusing the waiting here would make sense, I will add it - just double checking :)

Reply:
hm it depends on the expectations: do you expect that when this lines gets executed, the operator subscription must be ready? If yes, then it's ok not to wait

Reply:
Yes - my understanding is that this code is checking for an existing operator installation which had either:
- been installed some time ago (such as in previous build of the job) so the subscription is already ready,
- or the operator is not installed at all so the code would be just waiting while no installation had been requested.

The original code seemed to use this assumption, so I would vote for keeping it and if we see any issues, we can make it more resilient (and maybe slower). Would that be fine?